packages/graphql/mixin.server.js

```diff
@@ -82,14 +83,20 @@ class GraphQLMixin extends Mixin {
         });
   }
 
-  fetchData(data = {}, element) {
-    return this.prefetchData(element).then(() => data);
-  }
-
-  prefetchData(element) {
-    const prefetchOnServer = this.canPrefetchOnServer().every(value => value);
-
-    return prefetchOnServer ? getDataFromTree(element) : Promise.resolve();
+  async renderToFragments(element) {
+    if (this.canPrefetchOnServer().every(value => value)) {
+      let fragments = {};
+      const reactMarkup = await getMarkupFromTree({
+        tree: element,
+        renderFunction: tree => {
+          const { reactMarkup, ...renderedFragments } = renderToFragments(tree);
```

Comment:
Hm, IMO we don't need all that splatting and merging:
```js
  async renderToFragments(element) {
    if (this.canPrefetchOnServer().every(value => value)) {
      let fragments = {};
      await getMarkupFromTree({
        tree: element,
        renderFunction: tree => {
          fragments = renderToFragments(tree);
          return fragments.reactMarkup;
        },
      });
      return fragments;
    }
    return renderToFragments(element);
  }
```

And what do you think about returning early instead of nesting the happy path?
```js
  myFunction() {
    if (/* check for special case */) {
      // handling special case
      return;
    }

    // start of happy path
  }
```